File: code/HoChanny/week2/boj_14719.js
const fs = require('fs');
const inputData = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const board = inputData[1].split(' ').map((e) => +e);

let result = 0;

for(let i = 0; i < board.length; i++){
    let leftPillar = Math.max(...board.slice(0, i)); // 왼쪽에서 가장 높은 기둥 찾기
    let rightPillar = Math.max(...board.slice(i + 1)); // 오른쪽에서 찾기
    let standard = Math.min(leftPillar, rightPillar); // 둘 중 작은 값이 물이 고이는 최고 높이가 될 것이다

    if(board[i] < standard){
        result += standard - board[i];
    }
}

console.log(result);

File: code/jjuny-won/week2/boj_14719.py
h,w = map(int,input().split())
arr = list(map(int,input().split()))

#가장 큰 수를 기준으로 나눈다

result=0

# 범위를 w로 했을 경우 첫번째 요소의 left, 마지막 요소의 right를 구할 수 없기 때문에

for i in range(1,w-1):
    left = max(arr[:i])
    right = max(arr[i+1:])
    m = min(left,right)
    if(m>arr[i]):
        result += m-arr[i]
print(result)